### PR TITLE
feat(test): Migrating test_reg.py upstream

### PR DESCRIPTION
**Pull request by zpetrace**
Migrating test_reg.py upstream with slight changes to the original test

### CODE REVIEW

File: integration-tests/test_client.py [hunk 1]
```diff
@@ -97,3 +97,28 @@ def test_verify_logrotate_feature(insights_client):
...
+    directory = "/var/lib/insights"
+    registered_contents = [
+        "last_stable.egg",
+        "last_stable.egg.asc",
+    ]
+
+    insights_client.register()
+    dir_content_registered = [
+        os.path.basename(file) for file in glob.glob(f"{directory}/*")
+    ]
```
Comment (ptoscano) [inline integration-tests/test_client.py:116]: I think using `os.scandir()` to collect the filenames would be a better way to list the files here.

File: integration-tests/test_client.py [hunk 2]
```diff
@@ -97,3 +97,28 @@ def test_verify_logrotate_feature(insights_client):
...
+@pytest.mark.usefixtures("register_subman")
+def test_insights_directory_files(insights_client):
+    """
+    Test that /var/lib/insights have the right files available
+    """
+    directory = "/var/lib/insights"
+    registered_contents = [
+        "last_stable.egg",
+        "last_stable.egg.asc",
+    ]
```
Comment (ptoscano) [inline integration-tests/test_client.py:111]: Note that `newest.egg` (+ `.asc`) are also files that may show up in that directory.

Note²: it seems the "malware" application in insights-core stores `.json` cache files in that directory.

**Review by ptoscano — CHANGES_REQUESTED** [review]
To be honest, I'm a bit lost on what exactly is the test about: it seems to be asserting on the internals of `insights-core`, and in that case I'm not sure it is a good idea.

**Comment by zpetrace** [comment]
To answer the question  why this test, @ptoscano  - the original test_reg.py from the iqe test suite was designed to unregister, delete /var/lib/insights , re-register and see if the folder got created again but I know we don't usually want to delete directories during tests, right? - I remember that from one of your comments in former PRs
so what I was trying to do is to just see if the files that are expected to be there are there but if you'd prefer to have it the original way - deleting and checking if it gets created again I can definitely do that

File: integration-tests/test_client.py (hunk 2 above)
Comment (zpetrace) [reply integration-tests/test_client.py:111]: solved

File: integration-tests/test_client.py (hunk 1 above)
Comment (zpetrace) [reply integration-tests/test_client.py:116]: done

File: integration-tests/test_client.py [hunk 3]
```diff
@@ -115,3 +115,24 @@ def test_insights_details_file_exists(insights_client):
...
+@pytest.mark.usefixtures("register_subman")
+def test_insights_directory_files(insights_client):
+    """
+    Test that /var/lib/insights have content available
+    """
+    directory = "/var/lib/insights"
+    registered_contents = [
+        "last_stable.egg",
+        "last_stable.egg.asc",
+    ]
```
Comment (m-horky) [inline integration-tests/test_client.py:129]: Should this test also consider files like
- `/etc/insights-client/.lastupload`
- `/etc/insights-client/.registered`
 and so on? Those are always present on registered system as well.

Comment (zpetrace) [reply integration-tests/test_client.py:129]: Yeah but they are not in this directory, this test should focus just on that IMHO

**Review by m-horky — APPROVED** [review]
LGTM

**Pull request merged**